src/partition/ReceivedBoundingBox.cpp

```diff
@@ -0,0 +1,200 @@
+#include "partition/ReceivedBoundingBox.hpp"
+#include "com/CommunicateBoundingBox.hpp"
+#include "com/CommunicateMesh.hpp"
+#include "com/Communication.hpp"
+#include "utils/MasterSlave.hpp"
+#include "m2n/M2N.hpp"
+#include "utils/EventTimings.hpp"
+#include "mapping/Mapping.hpp"
+#include "mesh/Mesh.hpp"
+#include "utils/Helpers.hpp"
+#include "utils/MasterSlave.hpp"
+#include <vector>
+#include <map>
+
+using precice::utils::Event;
+
+namespace precice {
+namespace partition {
+
+ReceivedBoundingBox::ReceivedBoundingBox
+(
+  mesh::PtrMesh mesh, double safetyFactor, GeometricFilter geometricFilter)
+:
+  Partition (mesh),
+  _bb(mesh->getDimensions(), std::make_pair(std::numeric_limits<double>::max(), std::numeric_limits<double>::lowest())),
+  _dimensions(mesh->getDimensions()),
+  _safetyFactor(safetyFactor),
+  _geometricFilter(geometricFilter)
+{}
+
+void ReceivedBoundingBox::communicateBoundingBox()
+{  
+  prepareBoundingBox();
+  
+  if (not utils::MasterSlave::_slaveMode)
+  {
+    _remoteParComSize=0;    
+    _m2n->getMasterCommunication()->receive(_remoteParComSize, 0);
+
+    // we need to construct _globalBB first and then we can fill in that
+    mesh::Mesh::BoundingBox initialBB;
+    for (int i=0; i < _dimensions; i++) {
+      initialBB.push_back(std::make_pair(-1,-1));
+    }    
+    for (int remoteRank = 0; remoteRank < _remoteParComSize; remoteRank++ )
+    {
+      _globalBB[remoteRank]= initialBB;
+    }
+
+    // master receives global_bb from other master    
+    com::CommunicateBoundingBox(_m2n->getMasterCommunication()).receiveBoundingBoxMap(_globalBB, 0 );
+  }
+}
+
+void ReceivedBoundingBox::computeBoundingBox()
+{
+  // handle coupling mode first (i.e. serial participant)
+
+
+  if (not utils::MasterSlave::_slaveMode && not utils::MasterSlave::_masterMode)
+  { 
+    //some code here
```

Comment:
There is no code here.
```suggestion
    /// @todo implement
```